Makefile

```diff
@@ -63,6 +63,11 @@ codegen: ## Generate code. Must be run if changes are made to ./pkg/apis/...
 	# CRDs don't currently jive with VolatileTime, which has an Any type.
 	perl -pi -e 's/Any/string/g' charts/karpenter/templates/provisioning.karpenter.sh_provisioners.yaml
 	hack/boilerplate.sh
+	gen-crd-api-reference-docs \
```

Comment:
Very small nit on documentation versioning:

If we run `make codegen` before each PR, this means that we are modifying the end of the `docs/README.md` file to refer to this commit. If contributors run this in the future it may blur the line on when this doc actually changes (as this might get committed on each PR).

Reply:
Agreed -- however, it's already the case that these commits are bogus, since the act of generating them and commiting them result in different commit IDs. Because these commits are then squashed, it's already not accurate.

You should also only make codegen when you update the API files (or add new files entirely). We could potentially break out the boilerplate script into `verify` or something.